tests/Fixture/UniqueAuthorsFixture.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.2.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Test\Fixture;
+
+use Cake\TestSuite\Fixture\TestFixture;
+
+/**
+ * Tables of unique author ids
+ */
+class UniqueAuthorsFixture extends TestFixture
+{
+    /**
+     * fields property
+     *
+     * @var array
+     */
+    public $fields = [
+        'id' => ['type' => 'integer'],
+        'first_author_id' => ['type' => 'integer', 'null' => true],
+        'second_author_id' => ['type' => 'integer', 'null' => false],
+        '_constraints' => [
+            'primary' => ['type' => 'primary', 'columns' => ['id']],
+            'nullable_non_nullable_unique' => ['type' => 'unique', 'columns' => ['first_author_id', 'second_author_id']],
+        ],
+    ];
+
+    /**
+     * records property
+     *
+     * @var array
+     */
+    public $records = [
+        ['first_author_id' => null, 'second_author_id' => 1],
+    ];
```

Comment:
could we have 3 fields instead of  2 for testing ?

Reply:
Do you want the 3rd field to be NOT NULL?

Reply:
I want to test all these cases 

1)

| field_1| field_2| field_3 |  -
| ---| ---| ---| ---
| null| 1 | 1|  
 | null| 1 | 1| new record

2)

| field_1| field_2| field_3 |  -
| ---| ---| ---| ---
| null| 1 | 1|  
| null| 1 | 2| new record

3)

| field_1| field_2| field_3 |  -
| ---| ---| ---| ---
| null| 1 | 1|  
| null| null | 1| new record

4)

| field_1| field_2| field_3 |  -
| ---| ---| ---| ---
| null| 1 | 1|  
| null| null | 2| new record
| null| null | 2| new record

so when there are 2 `NULL` value , and when 2 `NOT NULL` value 
especially I want to see result of case 3 and 4 
you can add ` null| 1 | 1` in `Fixture`

Reply:
>The new code skips the check if there is a null value in the column set 

this is important for case 3

Reply:
I don't see the difference really. Why is 3 columns different than 2?